app/controllers/api/v1/posts_controller.rb

```diff
@@ -15,7 +15,9 @@ def show
       end
 
       def create
-        @post = Post.create(post_params)
+        @post = Post.new(post_params)
+        @post.user = current_user
+        @post.save!
```

Comment:
```suggestion
        @post.save
```
Иначе у тебя вывалится эксепшн и в итоге будет 500я ошибка вместо 422